test/smoke/generated/network-resource-manager/samples-dev/startPacketCaptureOnVpnGatewayWithFilter.ts

```diff
@@ -28,7 +28,9 @@ async function startPacketCaptureOnVpnGatewayWithFilter() {
     filterData:
       "{'TracingFlags': 11,'MaxPacketBufferSize': 120,'MaxFileSize': 200,'Filters': [{'SourceSubnets': ['20.1.1.0/24'],'DestinationSubnets': ['10.1.1.0/24'],'SourcePort': [500],'DestinationPort': [4500],'Protocol': 6,'TcpFlags': 16,'CaptureSingleDirectionTrafficOnly': true}]}"
   };
-  const options = { parameters: parameters };
+  const options = {
+    parameters: parameters as VpnGatewayPacketCaptureStartParameters
```

Comment:
Why is this cast needed? is there any way to make the types compatible without forcing a cast?

Reply:
Actually this is to mainly fix issues like this 
```
enum A {
    "AAA",
    "BBB"
};
const aa: A = "AAA";
console.log(aa);
```
Then the typescript compiler will complain about Type '"AAA"' is not assignable to type 'A'. And since we get the value from swagger example, we can only get the string type. I also don't think adding cast logic everywhere look like a good idea. just wondering what's the better way of resolve it?

Reply:
I'm not sure I follow the relationship of this change with enums. The cast should no be needed in this case or in any other instances this cast appears in this PR.

This line already makes parameters be type `VpnGatewayPacketCaptureStartParameters` so the cast is not needed.
```typescript
const parameters: VpnGatewayPacketCaptureStartParameters = {
```

Also please generate this line as 

```diff
- const options = {
-    parameters: parameters as VpnGatewayPacketCaptureStartParameters
- }
+ const options = { parameters }
```

Reply:
@joheredi I have updated the code to remove the casting usage in our samples and instead I add a type declaration for options parameter. which could also resolve the case I am facing with vmwarecloudsimple. 
You may use the following command with autorest.typescript main branch to repro the original issue if you are interested
```
autorest --version=3.7.3 --typescript --modelerfour.lenient-model-deduplication --head-as-boolean=true --license-header=MICROSOFT_MIT_NO_VERSION --generate-test --typescript-sdks-folder=. ../azure-rest-api-specs/specification/vmwarecloudsimple/resource-manager/readme.md --use=../autorest.typescript --generate-sample=true 
```